ccmlib/utils/ssl_utils.py

```diff
@@ -0,0 +1,63 @@
+    """
+
+    if os.path.exists(os.path.join(base_dir, 'keystore.jks')):
+        print("keystores already exists - skipping generation of ssl keystores")
+        return
+
+    legacy = '-legacy' if '-legacy' in subprocess.run(['openssl', 'pkcs12', '--help'],
+                                                      universal_newlines=True, stderr=subprocess.PIPE).stderr else ''
+    dns_names = dns_names or ['any.cluster-id.scylla.com']
+    ext = ",".join(["dns:{}".format(name) for name in dns_names])
+    print("generating keystore.jks in [{0}]".format(base_dir))
+    subprocess.check_call(['keytool', '-genkeypair', '-alias', 'ccm_node', '-keyalg', 'RSA', '-validity', '365',
+                           '-keystore', os.path.join(base_dir, 'keystore.jks'), '-storepass', passphrase,
+                           '-dname', 'cn=Cassandra Node,ou=CCMnode,o=DataStax,c=US', '-keypass', passphrase,
+                           '-ext', 'san={}'.format(ext)])
+
+    print("exporting cert from keystore.jks in [{0}]".format(base_dir))
+    subprocess.check_call(['keytool', '-export', '-rfc', '-alias', 'ccm_node',
+                           '-keystore', os.path.join(base_dir, 'keystore.jks'),
+                           '-file', os.path.join(base_dir, 'ccm_node.cer'), '-storepass', passphrase])
+    print("importing cert into truststore.jks in [{0}]".format(base_dir))
+    subprocess.check_call(['keytool', '-import', '-file', os.path.join(base_dir, 'ccm_node.cer'),
+                           '-alias', 'ccm_node', '-keystore', os.path.join(base_dir, 'truststore.jks'),
+                           '-storepass', passphrase, '-noprompt'])
+    # Added for scylla: Generate pem format cert/key
+    print("exporting cert to pks12 from keystore.jks in [{0}]".format(base_dir))
+    subprocess.check_call(['keytool', '-importkeystore', '-srckeystore', os.path.join(base_dir, 'keystore.jks'),
+                           '-srcstorepass', passphrase, '-srckeypass', passphrase, '-destkeystore',
+                           os.path.join(base_dir, 'ccm_node.p12'), '-deststoretype', 'PKCS12',
+                           '-srcalias', 'ccm_node', '-deststorepass', passphrase, '-destkeypass', passphrase])
+    print("Using openssl to split pks12 in [{0}] to pem format".format(base_dir))
+    subprocess.check_call(['openssl', 'pkcs12', '-in', os.path.join(base_dir, 'ccm_node.p12'),
+                           '-passin', 'pass:{0}'.format(passphrase), '-nokeys',
+                           '-out', os.path.join(base_dir, 'ccm_node.pem'), legacy])
+    # Key with password. We want without...
+    subprocess.check_call(['openssl', 'pkcs12', '-in', os.path.join(base_dir, 'ccm_node.p12'),
+                           '-passin', 'pass:{0}'.format(passphrase),
+                           '-passout', 'pass:{0}'.format(passphrase), '-nocerts',
+                           '-out', os.path.join(base_dir, 'ccm_node.tmp'), legacy])
```

Comment:
My OpenSSL on Fedora 34 (`openssl version: OpenSSL 1.1.1n  FIPS 15 Mar 2022`) does not like an empty argument (`legacy = ""`):

```
Traceback (most recent call last):
  File "/home/piotrgrabowski/scylla1/scylla-ccm/ccmlib/utils/ssl_utils.py", line 63, in <module>
    generate_ssl_stores('/home/piotrgrabowski/scylla1/scylla-ccm/ccmlib/utils/test1', dns_names=['any.cluster-id.scylla.com'])
  File "/home/piotrgrabowski/scylla1/scylla-ccm/ccmlib/utils/ssl_utils.py", line 49, in generate_ssl_stores
    subprocess.check_call(['openssl', 'pkcs12', '-in', os.path.join(base_dir, 'ccm_node.p12'),
  File "/usr/lib64/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['openssl', 'pkcs12', '-in', '/home/piotrgrabowski/scylla1/scylla-ccm/ccmlib/utils/test1/ccm_node.p12', '-passin', 'pass:cassandra', '-nokeys', '-out', '/home/piotrgrabowski/scylla1/scylla-ccm/ccmlib/utils/test1/ccm_node.pem', '']' returned non-zero exit status 1.
```

Removing `legacy` from the list (even though `legacy` is empty) makes the command run correctly.

Proposed fix:
```python
legacy = ['-legacy'] if '-legacy' in subprocess.run(['openssl', 'pkcs12', '--help'],
                                                  universal_newlines=True, stderr=subprocess.PIPE).stderr else []
...
subprocess.check_call(['openssl', 'pkcs12', '-in', os.path.join(base_dir, 'ccm_node.p12'),
                       '-passin', 'pass:{0}'.format(passphrase),
                       '-passout', 'pass:{0}'.format(passphrase), '-nocerts',
                       '-out', os.path.join(base_dir, 'ccm_node.tmp')] + legacy)
```

Reply:
😭 I'll fix it.